File: ios/Plugin/ResponseGenerator.swift
import Foundation

struct ResponseGenerator {

    private static let VALUE_RESPONSE_KEY = "value"
    private static let STATUS_RESPONSE_KEY = "status"

    static func fromBoolean(_ value: Bool) -> [String: Bool] {
        return value ? successResponse() : failResponse()
    }

    static func successResponse() -> [String: Bool] {
        return [VALUE_RESPONSE_KEY: true]
    }

    static func failResponse() -> [String: Bool] {
        return [VALUE_RESPONSE_KEY: false]
    }

    static func dataResponse(_ data: Any) -> [String: Any] {
        return [VALUE_RESPONSE_KEY: data]
    }

    static func statusResponse(_ data: CurrentRecordingStatus) -> [String: String] {
        return [STATUS_RESPONSE_KEY: data.rawValue]
    }

}

File: android/src/main/java/com/tchvu3/capacitorvoicerecorder/ResponseGenerator.java
package com.tchvu3.capacitorvoicerecorder;

import com.getcapacitor.JSObject;

public class ResponseGenerator {

    private static final String VALUE_RESPONSE_KEY = "value";
    private static final String STATUS_RESPONSE_KEY = "status";

    public static JSObject fromBoolean(boolean value) {
        return value ? successResponse() : failResponse();
    }

    public static JSObject successResponse() {
        JSObject success = new JSObject();
        success.put(VALUE_RESPONSE_KEY, true);
        return success;
    }

    public static JSObject failResponse() {
        JSObject success = new JSObject();
        success.put(VALUE_RESPONSE_KEY, false);
        return success;
    }

    public static JSObject dataResponse(Object data) {
        JSObject success = new JSObject();
        success.put(VALUE_RESPONSE_KEY, data);
        return success;
    }

    public static JSObject statusResponse(CurrentRecordingStatus status) {
        JSObject success = new JSObject();
        success.put(STATUS_RESPONSE_KEY, status.name());
        return success;
    }
}
